app/services/token_service.rb

```diff
@@ -3,12 +3,13 @@ class TokenService
   RANDOM_BYTES = 8
 
   class << self
-    def box(data)
+    def box(data, encryptor = nil)
+      encryptor = message_encryptor if encryptor.nil?
```

Comment:
```suggestion
      encryptor ||= message_encryptor
```

`message_encryptor` is memoized at the class level, seems pretty safe to me to make it a default param too

Reply:
It would be safe, yep. My concern with making it a default param is I think it kind of hides the complexity and how expensive it is. The cost is only paid the first time since it's memoized, but it's on the order of seconds to generate the key. For me, pulling it into the body signals it as a distinct step in the process.

Reply:
that makes sense. My next thought is that, if it's so expensive, should we make sure we "warm" that in an initializer?